provider/src/main/java/com/synopsys/integration/alert/database/api/BlackDuckProviderDataAccessor.java

```diff
@@ -0,0 +1,220 @@
+import com.synopsys.integration.rest.HttpUrl;
+
+@Component
+public class BlackDuckProviderDataAccessor implements ProviderDataAccessor {
+    private final IntLogger logger = new Slf4jIntLogger(LoggerFactory.getLogger(BlackDuckProviderDataAccessor.class));
+    private final ConfigurationAccessor configurationAccessor;
+    private final BlackDuckPropertiesFactory blackDuckPropertiesFactory;
+
+    @Autowired
+    public BlackDuckProviderDataAccessor(ConfigurationAccessor configurationAccessor, BlackDuckPropertiesFactory blackDuckPropertiesFactory) {
+        this.configurationAccessor = configurationAccessor;
+        this.blackDuckPropertiesFactory = blackDuckPropertiesFactory;
+    }
+
+    @Override
+    public List<ProviderProject> getProjectsByProviderConfigName(String providerConfigName) {
+        try {
+            Optional<ConfigurationModel> providerConfigOptional = configurationAccessor.getProviderConfigurationByName(providerConfigName);
+            if (providerConfigOptional.isPresent()) {
+                return getProjectsForProvider(providerConfigOptional.get());
+            }
+        } catch (IntegrationException e) {
+            logger.error(String.format("Could not get the project for the provider '%s'. %s", providerConfigName, e.getMessage()));
+            logger.debug(e.getMessage(), e);
+        }
+        return List.of();
+    }
+
+    @Override
+    public List<ProviderProject> getProjectsByProviderConfigId(Long providerConfigId) {
+        try {
+            Optional<ConfigurationModel> providerConfigOptional = configurationAccessor.getConfigurationById(providerConfigId);
+            if (providerConfigOptional.isPresent()) {
+                return getProjectsForProvider(providerConfigOptional.get());
+            }
+        } catch (IntegrationException e) {
+            logger.error(String.format("Could not get the project for the provider with id '%s'. %s", providerConfigId, e.getMessage()));
+            logger.debug(e.getMessage(), e);
+        }
+        return List.of();
+    }
+
+    private List<ProviderProject> getProjectsForProvider(ConfigurationModel blackDuckConfigurationModel) throws IntegrationException {
+        BlackDuckProperties properties = blackDuckPropertiesFactory.createProperties(blackDuckConfigurationModel);
+        BlackDuckHttpClient blackDuckHttpClient = properties.createBlackDuckHttpClient(logger);
+        BlackDuckServicesFactory blackDuckServicesFactory = properties.createBlackDuckServicesFactory(blackDuckHttpClient, logger);
+        ProjectService projectService = blackDuckServicesFactory.createProjectService();
+        List<ProjectView> allProjects = projectService.getAllProjects();
+        return convertBlackDuckProjects(allProjects, blackDuckServicesFactory.getBlackDuckService());
+    }
+
+    @Override
+    public void deleteProjects(Collection<ProviderProject> providerProjects) {
+        //ignored since we are not using the database
```

Comment:
For API's we have a response NOT_IMPLEMENTED. is there some similar runtime exception we could throw here?

Reply:
We don't want this to throw an exception. We want this class to be used in place of the DefaultProviderDataAccessor so that it will have minimal impact